nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/test/java/org/nuxeo/ecm/platform/ec/notification/TestEmailNotification.java

```diff
@@ -0,0 +1,198 @@
+
+        // subscribe to the dummy notification
+        addSubscription(documentModel);
+
+        // fire the dummy event
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), documentModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        // check the received mail
+        checkMailContent(documentModel, event);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToParentDocument() {
+        DocumentModel parentDocModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification on the parent document
+        addSubscription(parentDocModel);
+
+        // create a simple hierarchy (domain <- DOCUMENT_NAME <- ANOTHER_DOCUMENT_NAME)
+        DocumentModel mainDocModel = createDocument(parentDocModel, ANOTHER_DOCUMENT_NAME);
+
+        // fire the dummy event on the child document
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), mainDocModel);
+        Event event = ctx.newEvent(DUMMY_EVENT_NAME);
+        eventService.fireEvent(event);
+        transactionalFeature.nextTransaction();
+
+        // check the received mail
+        checkMailContent(mainDocModel, event);
+    }
+
+    protected void checkMailContent(DocumentModel documentModel, Event event) {
+        assertEquals(1, emailsResult.getMails().size());
+        SmtpMailServerFeature.MailMessage mailMessage = emailsResult.getMails().get(0);
+        // check the subject
+        assertEquals(String.format("[Dummy]Notification on the document '%s'", documentModel.getTitle()),
+                mailMessage.getSubject());
+
+        // check the text content
+        String expectedMailContent = getExpectedMailContent(documentModel, event);
+        assertEquals(expectedMailContent, getMailContent(mailMessage));
+    }
+
+    protected DocumentModel createDocument(DocumentModel parent, String name) {
+        DocumentModel documentModel = session.createDocumentModel(parent.getPathAsString(), name, "File");
+        return session.createDocument(documentModel);
+    }
+
+    protected void addSubscription(DocumentModel documentModel) {
+        NuxeoPrincipal principal = session.getPrincipal();
+        String subscriber = NotificationConstants.USER_PREFIX + principal.getName();
+        notificationManager.addSubscription(subscriber, DUMMY_NOTIFICATION_NAME, documentModel, false, principal,
+                DUMMY_NOTIFICATION_NAME);
+    }
+
+    protected String getMailContent(SmtpMailServerFeature.MailMessage mailMessage) {
+        String content = mailMessage.getContent();
+        if (!mailMessage.getContentType().contains("text/html")) {
+            return content;
+        }
+
+        Renderer renderer = new Source(mailMessage.getContent()).getRenderer();
```

Comment:
What's the point though? Can't you just compare the HTML strings?

Reply:
my idea was to create a html file and then use this method to extract the content without any html tag and to compare it to a text content. my point of view is to keep the things more closer to the real things where we have a html file (see https://github.com/nuxeo/nuxeo/pull/3705#discussion_r372356410)

my idea to not create a second html file, but if you think it  is better to have only html and compare them i will change it (i means i will create  a html expected content that will be compared to the received one)

Reply:
My point is that your goal here is not to unit test a HTML to text transformer, or FreeMarker generation. So you shouldn't care about either of those, and just compare expected to actual HTML.

Reply:
as asked i just use full html with the `ftl` once and i provide a new one that will be the expected once